### PR TITLE
Add fromBlock param to watch event functions

### PR DESCRIPTION
**Pull request by maxencerb**
Add the `fromBlock` params to `watchEvent` and `watchContractEvent` functions.

I needed to have a reliable, fast and easy way to start watching events from a specific block to build a storage reprensation of one of my contracts.

---

## PR-Codex overview
This PR adds the `fromBlock` parameter to `watchEvent` and `watchContractEvent` functions, allowing users to specify the block number to start listening for logs from.

### Detailed summary
- Added `fromBlock` parameter to `watchEvent` and `watchContractEvent`
- Updated function signatures in `watchContractEvent.ts` and `watchEvent.ts`
- Added tests for `fromBlock` parameter in `watchEvent.test.ts` and `watchContractEvent.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by jxom — COMMENTED** [review]
Can we please get some test coverage over these changes? Also a changeset would be nice! `bun run changeset`.

File: site/pages/docs/actions/public/watchEvent.md [hunk 1]
```diff
@@ -388,6 +388,23 @@ const unwatch = publicClient.watchEvent(
 )
 ```
 
+### fromBlock (optional)
+
+- **Type:** `bigint`
+
+The block number to start listenning for logs from.
```
Comment (jxom) [inline site/pages/docs/actions/public/watchEvent.md:395]:
```suggestion
The block number to start listening for logs from.
```

File: site/pages/docs/contract/watchContractEvent.md [hunk 2]
```diff
@@ -317,6 +317,21 @@ const unwatch = publicClient.watchContractEvent({
 })
 ```
 
+### fromBlock (optional)
+
+- **Type:** `bigint`
+
+The block number to start listenning for logs from.
```
Comment (jxom) [inline site/pages/docs/contract/watchContractEvent.md:324]:
```suggestion
The block number to start listening for logs from.
```

File: src/actions/public/watchContractEvent.ts [hunk 3]
```diff
@@ -88,6 +89,8 @@ export type WatchContractEventParameters<
    * @default false
    */
   strict?: strict | boolean | undefined
+  /** block to start listenning from */
```
Comment (jxom) [inline src/actions/public/watchContractEvent.ts:92]:
```suggestion
  /** Block to start listening from. */
```

File: src/actions/public/watchEvent.ts [hunk 4]
```diff
@@ -74,6 +75,8 @@ export type WatchEventParameters<
   onError?: ((error: Error) => void) | undefined
   /** The callback to call when new event logs are received. */
   onLogs: WatchEventOnLogsFn<TAbiEvent, TAbiEvents, TStrict, _EventName>
+  /** The block to start from */
```
Comment (jxom) [inline src/actions/public/watchEvent.ts:78]:
```suggestion
  /** Block to start listening from. */
```

**Comment by maxencerb** [comment]
Added a patch changeset and coverage for `fromBlock` on both modified functions.

**Pull request merged**